cupy/random/_generator.py

```diff
@@ -1072,6 +1072,8 @@ def choice(self, a, size=None, replace=True, p=None):
             if not isinstance(shape, int):
                 index = cupy.reshape(index, shape)
         else:
+            if a_size == 0:
+                a_size = 1
```

Comment:
This workaround does not seem needed if `randint` method is fixed, too.  Let's fix it in another PR.

Could you comment here that ``# TODO: (#4511) Fix `randint` instead``?